kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -0,0 +1,456 @@
+
+        if instances is not None and len(instances) > 1:
+            raise Exception('Kubelet check only supports one configured instance.')
+        inst = instances[0] if instances else None
+
+        self.kubelet_conn_info = get_connection_info()
+        if not self.kubelet_conn_info.get('url'):
+            raise Exception("Couldn't determine the kubelet URL, kubelet check won't proceed.")
+
+        self.kube_node_labels = inst.get('node_labels_to_host_tags', {})
+        self.pod_list_url = urljoin(self.kubelet_conn_info['url'], POD_LIST_PATH)
+        self.kube_health_url = urljoin(self.kubelet_api_url, KUBELET_HEALTH_PATH)
+        self.node_spec_url = urljoin(self.kubelet_api_url, NODE_SPEC_PATH)
+
+        self.metrics_mapper = {
+            'kubelet_runtime_operations_errors': 'kubelet.runtime.errors',
+        }
+        self.ignore_metrics = [
+            'container_cpu_cfs_periods_total',
+            'container_cpu_cfs_throttled_periods_total',
+            'container_cpu_cfs_throttled_seconds_total',
+            'container_cpu_load_average_10s',
+            'container_cpu_system_seconds_total',
+            'container_cpu_user_seconds_total',
+            'container_fs_inodes_free',
+            'container_fs_inodes_total',
+            'container_fs_io_current',
+            'container_fs_io_time_seconds_total',
+            'container_fs_io_time_weighted_seconds_total',
+            'container_fs_read_seconds_total',
+            'container_fs_reads_merged_total',
+            'container_fs_reads_total',
+            'container_fs_sector_reads_total',
+            'container_fs_sector_writes_total',
+            'container_fs_write_seconds_total',
+            'container_fs_writes_merged_total',
+            'container_fs_writes_total',
+            'container_last_seen',
+            'container_start_time_seconds',
+            'container_spec_memory_swap_limit_bytes',
+            'container_scrape_error'
+        ]
+
+        # these are filled by container_<metric-name>_usage_<metric-unit>
+        # and container_<metric-name>_limit_<metric-unit> reads it to compute <metric-name>usage_pct
+        self.fs_usage_bytes = {}
+        self.mem_usage_bytes = {}
+
+    def check(self, instance):
+        endpoint = instance.get('metrics_endpoint')
+        if endpoint is None:
+            raise CheckException("Unable to find metrics_endpoint in config file.")
+
+        send_buckets = instance.get('send_histograms_buckets', True)
+        # By default we send the buckets.
+        if send_buckets is not None and str(send_buckets).lower() == 'false':
+            send_buckets = False
+        else:
+            send_buckets = True
+
+        try:
+            pod_list = self.retrieve_pod_list()
+        except Exception:
+            pod_list = None
+
+        instance_tags = instance.get('tags', [])
+        self._report_node_metrics(instance_tags)
+        self._perform_kubelet_check(instance_tags)
```

Comment:
We should move `_perform_kubelet_check` to first position, it avoid running into an exception in `_report_node_metrics` before submitting the service check